### ISSUE TITLE
Editing a SASS partial in v0.20 does not trigger the output CSS file to be regenerated when "always enable controller" is false

### ISSUE THREAD

**Issue by sprak3000** [markstory/asset_compress#329]
Bear with me here... This takes a bit of setup to explain. We were using v0.13 and recently upgraded to v0.20. As we began to develop locally, we noticed our CSS updates did not seem to be picked up.

Our team has our local development environments `asset_compress.local.ini` configured this way:

```
[General]
; Keep cacheConfig at false for production where assets are generated during
; the build/deployment process. If set to true in PROD, the wrong full path
; are being cached causing errors.
cacheConfig = false
; You should probably set this to true locally so you don't have to regenerate
; asset everytime
alwaysEnableController = false
```
We use this configuration to have the files generated on every request. We use SASS and make use of partials. Here is a sample from `asset_compress.ini`:

```
[css_main.css]
files[] = main.css.scss
```
And a brief snippet from that SASS file:
```
@import "partials/notes";
@import "partials/tasks";

@import "partials/nprogress";

.import-link {
    font-size: 14px;
    padding-left: 15px;
}
...
```
If I edit `partials/notes`, the change does not seem to be applied. I dug into the code and found the reason.

`AssetCompressor::beforeDispatch()` eventually calls `AssetCache::isFresh()` passing in `main.css` as the "target". `isFresh()` takes `main.css` and says "Hey, the file list for this target only includes `main.css.scss`". Where that wasn't touched, it is marked as fresh, and `AssetCompressor::beforeDispatch()` decides to use a file previously cached on disk.

Comparing this flow to what was in v0.13 shows this is a different strategy than before. v0.13 seemed to be comparing data within the event passed into `beforeDispatch()` and using that to determine if the file should be rebuilt rather than going to the actual file list configured for the target asset.

I've been staring at this to see if could pull together a PR to go with this, but the change seems so fundamental a shift. Basically, `isFresh()` would somehow need to identify the file in the target's list of files is a SASS file, parse it for imports, and then check the freshness of the imports. But how far down the rabbit hole? Do the imports have imports we need to check?

Our team can work around this by either clearing the cache on disk or touching the SASS file, but I'm willing and hoping to brainstorm a possible solution to this. I also realize this version may no longer be supported. I'll look into the latest code to see if there is something I could back port but am hoping the author(s) might be able to shed some light on the change.

**Comment by markstory** [comment]
I don't think there will be much chance in getting fixes made to 0.13, as it is 4+ years old. To be honest my memory of why that behavior changed has been lost over the years.

The recursive import logic you're describing has been implemented in the 3.x release series. I've never made an effort to back-port it as I no longer maintain any CakePHP 2 applications. That code can be found [here](https://github.com/markstory/mini-asset/blob/master/src/Utility/CssUtils.php) and [here](https://github.com/markstory/mini-asset/blob/b24d51043f8aa003dc451e72e38561cdfe591056/src/Filter/CssDependencyTrait.php).

If you'd like to backport the dependency resolution it will be a pretty big change and you'd want to build off of the `2.x` branch and not off of `0.20` as that tag is almost 4 years old as well.

**Comment by sprak3000** [comment]
@markstory 

Thank you for pointing me to the code in the 3.X release. I've cobbled together a PR for the 2.X branch based on it. Bunch of code duplication going on in there, but I kept it simple for now to get things running.
